### PR TITLE
Make repeated WGS84-Mercator conversion stable

### PR DESCRIPTION
**Pull request by abyrd**
This PR supersedes #892, which has been split into multiple PRs. This is a more minimal changeset than originally proposed, to facilitate review and contain any risk from changing heavily used math functions.

As recently discussed: regional request bounds are specified in WGS84 coordinates, which are converted to web Mercator extents when regional tasks are processed. Repeating a regional analysis with the same bounds as an existing one requires the the UI to look at the web Mercator bounds of an existing analysis, then convert them to WGS84 to create the new regional analysis request. This creates a feedback loop between WGS84 and web Mercator where WGS84 bounding boxes lie exactly on the edges of Mercator pixels, creating a literal edge case for the function `WebMercatorExtents#forWgsEnvelope` and `populateTask`.

This PR revises the way web Mercator extents are created to handle this edge case. This change is reinforced with additional factory methods that should handle any numerical instability or lack of precision in the repeated conversions.

The eventual real fix for this is https://github.com/conveyal/r5/issues/893.

This PR includes a test of repeated conversions, which fails without these changes and passes with these changes. 
Javadoc was also added and updated.

### CODE REVIEW

File: src/main/java/com/conveyal/r5/analyst/WebMercatorExtents.java [hunk 1]
```diff
@@ -114,23 +118,82 @@ public WebMercatorExtents expandToInclude (WebMercatorExtents other) {
...
+        // Note that web Mercator coordinates increase from north to south, so maximum latitude is minimum Mercator y.
         int north = latToPixel(wgsEnvelope.getMaxY(), zoom);
         int west = lonToPixel(wgsEnvelope.getMinX(), zoom);
-        int height = (latToPixel(wgsEnvelope.getMinY(), zoom) - north) + 1; // minimum height is 1
-        int width = (lonToPixel(wgsEnvelope.getMaxX(), zoom) - west) + 1; // minimum width is 1
-        WebMercatorExtents webMercatorExtents = new WebMercatorExtents(west, north, width, height, zoom);
-        return webMercatorExtents;
+        // Find width and height in whole pixels, handling the case where the right envelope edge is on a pixel edge.
+        int height = (int) Math.ceil(latToFractionalPixel(wgsEnvelope.getMinY(), zoom) - north);
+        int width = (int) Math.ceil(lonToFractionalPixel(wgsEnvelope.getMaxX(), zoom) - west);
```
Comment (abyrd) [inline src/main/java/com/conveyal/r5/analyst/WebMercatorExtents.java:141]: Note that this is a key part of the PR, using ceil() to handle the case where the projected value already lies exactly on a pixel boundary.

**Review by ansoncfit — APPROVED** [review]
These changes look good to me.

**Comment by abyrd** [comment]
A corresponding or related UI PR is https://github.com/conveyal/ui/pull/1985, where I made some comments about coordinating behavior between the two sides.

**Review by trevorgerhardt — APPROVED** [review]
👍

**Pull request merged**